src/tests/ftest/util/dmg_utils.py

```diff
@@ -44,72 +44,266 @@ def __init__(self, path):
 
-    class DmgFormatSubCommand(CommandWithParameters):
-        """Defines an object representing a format sub dmg command."""
+    class NetworkSubCommand(CommandWithSubCommand):
+        """Defines an object for the dmg network sub command."""
 
         def __init__(self):
-            """Create a dmg Command object."""
-            super(DmgCommand.DmgFormatSubCommand, self).__init__(
-                "/run/dmg/format/*", "format")
-            self.reformat = FormattedParameter("--reformat", False)
-
-    class DmgPrepareSubCommand(CommandWithParameters):
-        """Defines a object representing a prepare sub dmg command."""
+            """Create a dmg network subcommand object."""
+            super(DmgCommand.NetworkSubCommand, self).__init__(
+                "/run/dmg/network/*", "network")
+
+        def get_sub_command_class(self):
+            """Get the dmg network sub command object."""
+            if self.sub_command.value == "scan":
+                self.sub_command_class = self.ScanSubCommand()
+            else:
+                self.sub_command_class = None
+
+        class ScanSubCommand(CommandWithParameters):
+            """Defines an object for the dmg network scan command."""
+
+            def __init__(self):
+                """Create a dmg network scan command object."""
+                super(
+                    DmgCommand.NetworkSubCommand.ScanSubCommand, self).__init__(
+                        "/run/dmg/network/scan/*", "scan")
+                self.provider = FormattedParameter("-p {}", None)
+                self.all = FormattedParameter("-a", False)
+
+    class PoolSubCommand(CommandWithSubCommand):
+        """Defines an object for the dmg pool sub command."""
 
         def __init__(self):
-            """Create a dmg Command object."""
-            super(DmgCommand.DmgPrepareSubCommand, self).__init__(
-                "/run/dmg/prepare/*", "prepare")
-            self.pci_wl = FormattedParameter("-w {}")
-            self.hugepages = FormattedParameter("-p {}")
-            self.targetuser = FormattedParameter("-u {}")
-            self.nvmeonly = FormattedParameter("-n", False)
-            self.scmonly = FormattedParameter("-s", False)
-            self.force = FormattedParameter("-f", False)
-            self.reset = FormattedParameter("--reset", False)
-
-    class DmgCreateSubCommand(CommandWithParameters):
-        """Defines a object representing a create sub dmg command."""
+            """Create a dmg pool subcommand object."""
+            super(DmgCommand.PoolSubCommand, self).__init__(
+                "/run/dmg/pool/*", "pool")
+
+        def get_sub_command_class(self):
+            """Get the dmg pool sub command object."""
+            if self.sub_command.value == "create":
+                self.sub_command_class = self.CreateSubCommand()
+            elif self.sub_command.value == "delete-acl":
+                self.sub_command_class = self.DeleteAclSubCommand()
```

Comment:
(pylint-redefined-variable-type) Redefinition of self.sub_command_class type from util.dmg_utils.DmgCommand.PoolSubCommand.CreateSubCommand to util.dmg_utils.DmgCommand.PoolSubCommand.DeleteAclSubCommand